### ISSUE TITLE
Fix `function-url-quotes` false positives when URLs require quoting

### ISSUE THREAD

**Issue by tbradsha** [stylelint/stylelint#8544]
### What minimal example or steps are needed to reproduce the bug?

```css
a {
	background-image: url('data:image/svg+xml;utf8,<svg height="100" width="100" xmlns="http://www.w3.org/2000/svg"><circle r="45" cx="50" cy="50" fill="red"/></svg>');
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "function-url-quotes": "never"
  }
}
```

### How did you run Stylelint?

[Demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYAdAdugLgIyTAGsBzAJwgFdMoBaASwFskSZ4VKyAbACgHIoSHEnhMWMAPQBnAG4kA1AA9GXANyUcAMwAcAGgA8skigAWMeiRM4AvOhABGAAyO7KAO70oOE7YfPXylyYUr5WOAAO8BISbrEAdG4AzHEQZCQSAEzOjtJydgB8+mD0ZGBcMChkvgAsAKyuYIq+tS4gKGAAns2tKJr0XFy+ZDBQdhKFuST5fACUqlgAvlgguiB95QBySIxwiDCK2+HlcWBSUivgEJh9JAggGJgoKHZklOVSduwPT092mtRgHD0K60ThcWgAR0oEBwMA+CGeIEwMBkMDIdiwTyWmAWF0g1wsADFUswcHcAFZSK4XWDhc6Ib6IqQ4DrlLj0TA4T6IrhCOFclaYpkstkcnC0fE3WjMpA0JBkUYIuy82HMjE4kALIA)

### Which Stylelint-related dependencies are you using?

```json
{
  "stylelint": "latest",
  "stylelint-config-standard": "latest"
}
```

### What did you expect to happen?

No errors and no fix applied.

### What actually happened?

```
2:24-159 ERROR Unexpected quotes around "url" function argument (function-url-quotes)
```

### Do you have a proposal to fix the bug?

Some inspiration could be taken from `font-family-name-quotes`.

In instances where the `url()` content has quotes or spaces inside, it'd be best to leave it alone by default.

Alternatively or in addition, a secondary option might be adding `ignoreURL`, ignoring the contents if it matches a specific pattern.

**Comment by tbradsha** [comment]
I noticed another oddity with this rule:

1. This is a simple example triggers the error:
```
a {
	background-image: url( 'data:image/svg+xml,<svg viewBox="0 0 10 10"></svg>' );
}
```

2. This also triggers the error:
```
a {
	background-image: url( 'data:image/svg+xml,<svg viewBox="0 0 10 10"><style>.cls-1{}</style></svg>' );
}
```

3. However, this does NOT trigger the error. I suspect that's due to a different parsing issue.
```
a {
	background-image: url( 'data:image/svg+xml,<svg viewBox="0 0 10 10"><style>.cls-1{fill-rule:evenodd;}</style></svg>' );
}
```

**Comment by jeddy3** [comment]
@tbradsha Thanks for the report and reproducible demo.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to fix a bug in a rule](https://stylelint.io/developer-guide/rules#fix-a-bug-in-a-rule) in the Developer guide.

**Comment by romainmenke** [comment]
I think this case could be fine to report and autofix.

If proper value escaping was applied to the examples the url token would work, even without quotes:

```css
body {
	background-image: url( data:image/svg+xml,<svg\ viewBox=\"0\ 0\ 10\ 10\"></svg> );
}
```

So maybe we should ensure that any value is always properly escaped?

**Comment by tbradsha** [comment]
> So maybe we should ensure that any value is always properly escaped?

Indeed, for the "always" option this is probably the best option if there's a consistent set of rules to do this.

Adding an "always when recommended" option that skips quote removal in these scenarios would be my preference as a default, as escaping everything can get quite messy.

For maximum user-end flexibility, a user-defined "ignore" secondary option would be good, too.

**Comment by taearls** [comment]
Hey! I've opened a [PR](https://github.com/stylelint/stylelint/pull/8804) that addresses this issue.

**Changes made:**
- Added test cases covering the bug (unquoted URLs with special characters that require quotes)
- Fixed the rule to not report errors when quotes are required in `url()` functions

**Scope note:** This PR focuses on fixing the false positive when quotes are required. It does not add validation for proper escaping of URL values.

Let me know if you'd like me to adjust anything!
